encoding/multi_reader_iterator.go

```diff
@@ -169,7 +169,7 @@ func (it *multiReaderIterator) moveIteratorToValidNext(iter Iterator) bool {
 	}
 
 	err := iter.Err()
-	if it.err == nil && err != nil {
+	if it.err == nil && err != nil && err != io.EOF {
```

Comment:
As far as our use case is concerned, the underlying reader iterators should never run out of bytes because the encoded byte stream associated with each iterator should always end with an end-of-stream marker (see https://github.com/m3db/m3db/blob/master/encoding/scheme.go#L32), at which point the iterator knows it should stop reading bytes from the stream. IOW, an EOF error from the underlying stream is indicative of an underlying data issue.